marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/RowBatcherImpl.java

```diff
@@ -0,0 +1,312 @@
+package com.marklogic.client.datamovement.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.*;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.impl.BatchPlanImpl;
+import com.marklogic.client.impl.DatabaseClientImpl;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.io.marker.StructureReadHandle;
+import com.marklogic.client.util.RequestParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class RowBatcherImpl extends BatcherImpl implements RowBatcher {
+
+    private int batchSize;
+    private  int threadCount;
+    private ForestConfiguration forestConfig;
+    private String jobId;
+    private  String jobName;
+    private JobTicket jobTicket;
+    private Calendar jobStartTime;
+    private Calendar jobEndTime;
+    private PlanBuilder.PreparePlan preparePlan;
+    private DatabaseClient client;
+    private AtomicLong lowerBound = new AtomicLong(0);
+    private long maxBatches = Long.MAX_VALUE;
+    private long rowCount;
+    private Long bucketSize;
```

Comment:
I'm wondering whether the bucket size can be an integer -- that is, whether we would ever retrieve more rows in a single request than Integer.MAX_VALUE.